src/Servers/Kestrel/Transport.Libuv/test/LibuvOutputConsumerTests.cs

```diff
@@ -740,14 +741,16 @@ private LibuvOuputProcessor CreateOutputProducer(PipeOptions pipeOptions, Cancel
             var connectionFeatures = new FeatureCollection();
             connectionFeatures.Set(Mock.Of<IConnectionLifetimeFeature>());
 
+            var concurrentPipeWriter = new ConcurrentPipeWriter(pair.Transport.Output, _memoryPool);
```

Comment:
I had to wrap the DefaultPipeWriter with ConcurrentPipeWriter in LibuvOutputConsumerTests since TimingPipeFlusher now no longer protects against concurrent flushes. This is way the tests weren't completing with earlier iterations of this PR.

I think this logic that I added to the test to wrap the passed-in PipeWriter with the ConcurrentPipeWriter should just be moved into Http1OutputProducer to avoid regressing our old flushing behavior.

This does mean using the ConcurrentPipeWriter for both HTTP/1 and HTTP/2. Does everyone agree that's OK? @davidfowl

Reply:
An alternative would be to revert the changes to TimingPipeFlusher to remove the Task queu that were made in this PR, but I don't like that as much.

Reply:
No I think using this in both HTTP/1 and HTTP/2 is fine but then we need to measure the performance to make sure we don't regress anything.